### PR TITLE
Issue #7637: Update doc for EqualsAvoidNull

### PR DESCRIPTION
**Pull request by AmrDeveloper**
Issue #7637

pr1

pr2

Example 1:
```
cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
       <module name="EqualsAvoidNull"/>
    </module>
</module>

cat Test.java
package main;

public class Test {
    public static void main(String[] args) {
        String nullString = null;
        nullString.equals("My_Sweet_String");            // violation
        "My_Sweet_String".equals(nullString);            // OK
        nullString.equalsIgnoreCase("My_Sweet_String");  // violation
        "My_Sweet_String".equalsIgnoreCase(nullString);  // OK
    }
}

java -jar checkstyle-8.36.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] D:\OpenSource\TestCommit\Test.java:6:26: String literal expressions should be on the left side of an equals comparison. [EqualsAvoidNull]
[ERROR] D:\OpenSource\TestCommit\Test.java:8:36: String literal expressions should be on the left side of an equalsIgnoreCase comparison. [EqualsAvoidNull]
Audit done.
Checkstyle ends with 2 errors.
```
Example 2:

```
cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
       <module name="EqualsAvoidNull">
           <property name="ignoreEqualsIgnoreCase" value="true"/>
       </module>
    </module>
</module>

cat Test.java
package main;

public class Test {
    public static void main(String[] args) {
        String nullString = null;
        nullString.equals("My_Sweet_String");            // violation
        "My_Sweet_String".equals(nullString);            // OK
        nullString.equalsIgnoreCase("My_Sweet_String");  // OK
        "My_Sweet_String".equalsIgnoreCase(nullString);  // O 
    }
}

java -jar checkstyle-8.36.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] D:\OpenSource\TestCommit\Test.java:6:26: String literal expressions should be on the left side of an equals comparison. [EqualsAvoidNull]
Audit done.
Checkstyle ends with 1 errors.
```

I have removed the example from the Description section and moved it with other examples in the Examples section

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java [hunk 1]
```diff
@@ -38,23 +38,8 @@
  * </p>
  * <p>Rationale: Calling the {@code equals()} method on String literals
  * will avoid a potential {@code NullPointerException}. Also, it is
- * pretty common to see null checks right before equals comparisons,
- * which is not necessary in the example below.
+ * pretty common to see null checks right before equals comparisons.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java:41]: `pretty common to see null checks right before equals comparisons but following this rule such checks are not required`.

**Review by romani — CHANGES_REQUESTED** [review]
item to improve:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java (hunk 1 above)
Comment (AmrDeveloper) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java]: @romani Done

**Review by romani — APPROVED** [review]

**Comment by romani** [comment]
@AmrDeveloper, thanks a lot, welcome back.

**Comment by AmrDeveloper** [comment]
@romani Thanks

**Pull request merged**